contracts/contracts/IWarpMessenger.sol

```diff
@@ -0,0 +1,53 @@
+// (c) 2022-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+struct WarpMessage {
+    bytes32 originChainID;
+    bytes32 originSenderAddress;
+    bytes32 destinationChainID;
+    bytes32 destinationAddress;
+    bytes payload;
+}
+
+interface WarpMessenger {
+    event SendWarpMessage(
+        bytes32 indexed destinationChainID,
+        bytes32 indexed destinationAddress,
+        bytes32 indexed sender,
+        bytes message
+    );
+
+    // sendWarpMessage emits a request for the subnet to send a warp message from [msg.sender]
+    // with the specified parameters.
+    // This emits a SendWarpMessage log, which will be picked up by validators to queue the signing of
```

Comment:
This is intentionally `validators` not relayers. The validators handle the produced log on accept and add it to the warp backend, so that they are willing to sign it.

I'll re-phrase this to make it more clear that validators are willing to sign and aggregating signatures is left to relayers.